algebra/src/to_field_vec.rs

```diff
@@ -8,29 +8,47 @@ use crate::{
         short_weierstrass_jacobian::Jacobian,
         short_weierstrass_projective::Projective,
         twisted_edwards_extended::TEExtended,
-    },
+    }, ToBits,
 };
 
 type Error = Box<dyn std::error::Error>;
 
+macro_rules! slice_to_constraint_field {
+    ($base: ident) => {
+        impl<'a, F: Field> ToConstraintField<F> for &'a [$base] {
+            fn to_field_elements(&self) -> Result<Vec<F>, Error> {
+                ToConstraintField::<F>::to_field_elements(*self)
+            }
+        }
+    };
+}
+
 /// Types that can be converted to a vector of `F` elements. Useful for specifying
 /// how public inputs to a constraint system should be represented inside
 /// that constraint system.
 pub trait ToConstraintField<F: Field> {
     fn to_field_elements(&self) -> Result<Vec<F>, Error>;
 }
 
-impl<F: PrimeField> ToConstraintField<F> for F {
+impl<F: Field, T: ToConstraintField<F> + Clone, const N: usize> ToConstraintField<F> for [T; N] {
     fn to_field_elements(&self) -> Result<Vec<F>, Error> {
-        Ok(vec![*self])
+        self.to_vec().to_field_elements()
     }
```

Comment:
I try to avoid as much as possible these kind of implementations.
Unfortunately it's not possible to do it as ToConstraintField is already implemented for a generic template (F) so we cannot have more implementations of this type (compiler will give you an error like "conflicting implementation")

Reply:
Let's duplicate the code, it's not much anyway